include/tscpp/api/TransformationPlugin.h

```diff
@@ -97,13 +98,19 @@ class TransformationPlugin : public TransactionPlugin
 
   /**
    * Call this method if you wish to pause the transformation.
-   * Schedule the return value continuation to resume the transforamtion.
-   * If the continuation is scheduled and called after the transform is destroyed it
-   * won't do anything beyond cleanups.
-   * Note: You must schedule the continuation or destroy it (using TSContDestroy) yourself,
-   * otherwise it will leak.
    */
-  TSCont pause();
+  void pause();
+
+  /**
+   * Returns true if the transformation is paused, false otherwise.
+   */
+  bool isPaused() const;
+
+  /**
+   * Returns a reference to the Continuation to schedule to resume (unpause) the transformation.  It will be resumed when this
+   * continuation is executed.  Must not call this unless the transformation is paused.
+   */
+  Continuation &resumeCont();
```

Comment:
Not sure I like this, the asymmetry bothers me. It seems like it should be `resume` and if necessary it should take the delay and thread pool. If `pause` hides the continuation, then so should `resume`.

Reply:
I think your are assuming that the Continuation would be executed with the schedule() member function.  Are you sure there would never be a reason to execute it with the call() or httpSchedule() member funcs?

Reply:
using `HttpSchedule` seems very unlikely, but `TSContCall` is more plausible. Thinking about that, I would be more tempted to have resume

*  Get the lock for the resume continuation and try to lock.
*  If the lock is acquired, `TSContCall`
*  If not, schedule immediate.

That's the best option and realistically I don't see a use case for a different approach.

Reply:
OK should the resume() function take a timeout and a thread pool parameter, and should there be any defaults?

Reply:
TSContCall calls Continuation::dispatchEvent():  https://github.com/apache/trafficserver/blob/867d9c8527117ba0ac61fc2e282741e720b78cf8/iocore/eventsystem/Continuation.cc#L28

Does this function already do exactly what you said?  Why should I want to do it twice?  Does **anyone** really know how this crap works?